File: backend/src/main/java/com/ufc/backend/backend/model/based/FeedbackBased.java
package com.ufc.backend.backend.model.based;

import com.ufc.backend.backend.model.feedback.Feedback;
import lombok.*;
import org.springframework.data.mongodb.core.mapping.DBRef;
import java.util.List;

@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public abstract class FeedbackBased implements FeedbackBasedInterface {
    @DBRef
    private List<Feedback> feedbacks;

    @Override
    public List<Feedback> findAllFeedbacks() {
        return this.feedbacks;
    }

    @Override
    public void addFeedback(Feedback feedback) {
        if (this.feedbacks == null)
            setFeedbacks(List.of(feedback));
        else
            this.findAllFeedbacks().add(feedback);
    }

    @Override
    public void deleteFeedback(Feedback feedback) {
        this.findAllFeedbacks().removeIf(obj -> obj.getId().equals(feedback.getId()));
    }

    @Override
    public void updateFeedback(Feedback newFeedback) {
        deleteFeedback(newFeedback);
        addFeedback(newFeedback);
    }
}

File: backend/src/main/java/com/ufc/backend/backend/model/based/FeedbackBased.kt
package com.ufc.backend.backend.model.based

import com.ufc.backend.backend.model.feedback.Feedback
import lombok.AllArgsConstructor
import lombok.NoArgsConstructor
import lombok.Setter
import lombok.ToString
import org.springframework.data.mongodb.core.mapping.DBRef

@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
abstract class FeedbackBased : FeedbackBasedInterface {
    @DBRef
    private val feedbacks: MutableList<Feedback> = ArrayList()
    override fun findAllFeedbacks(): MutableList<Feedback> {
        return feedbacks
    }

    override fun addFeedback(feedback: Feedback) {
        findAllFeedbacks().add(feedback)
    }

    override fun deleteFeedback(feedback: Feedback) {
        findAllFeedbacks().remove(feedback)
    }

    override fun updateFeedback(newFeedback: Feedback) {
        deleteFeedback(newFeedback)
        addFeedback(newFeedback)
    }
}